File: go-client-sdk/pkg/models/shared/codatclientsapiclientcontractsyncsetting.go
package shared

import (
	"time"
)

type CodatClientsAPIClientContractSyncSetting struct {
	DataType         *string    `json:"dataType,omitempty"`
	FetchOnFirstLink *bool      `json:"fetchOnFirstLink,omitempty"`
	MonthsToSync     *int       `json:"monthsToSync,omitempty"`
	SyncFromUtc      *time.Time `json:"syncFromUtc,omitempty"`
	SyncFromWindow   *int       `json:"syncFromWindow,omitempty"`
	SyncOrder        *int       `json:"syncOrder,omitempty"`
	SyncSchedule     *int       `json:"syncSchedule,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatclientsapiclientcontractsyncsetting.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";


export class CodatClientsApiClientContractSyncSetting extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=dataType" })
  dataType?: string;

  @SpeakeasyMetadata({ data: "json, name=fetchOnFirstLink" })
  fetchOnFirstLink?: boolean;

  @SpeakeasyMetadata({ data: "json, name=monthsToSync" })
  monthsToSync?: number;

  @SpeakeasyMetadata({ data: "json, name=syncFromUtc" })
  syncFromUtc?: Date;

  @SpeakeasyMetadata({ data: "json, name=syncFromWindow" })
  syncFromWindow?: number;

  @SpeakeasyMetadata({ data: "json, name=syncOrder" })
  syncOrder?: number;

  @SpeakeasyMetadata({ data: "json, name=syncSchedule" })
  syncSchedule?: number;
}